rule/expr.go

```diff
@@ -209,7 +206,7 @@ func stringValue(e bzl.Expr) string {
 // The four collections may appear in any order, and some or all of them may
 // be omitted (all fields are nil for a nil expression).
 type platformStringsExprs struct {
-	generic            *bzl.ListExpr
+	generic            bzl.Expr
```

Comment:
This needs to stay `*bzl.ListExpr`. `platformStringsExprs` is an expression matching a specific pattern (described in the doc comment). The list expression cannot be anything other than a list.